parsl/usage_tracking/usage.py

```diff
@@ -192,7 +192,17 @@ def send_message(self) -> None:
 
         self.send_UDP_message(message)
 
-    def close(self) -> None:
-        """We terminate (SIGTERM) the processes added to the self.procs list """
+    def close(self, timeout: float = 10.0) -> None:
+        """First give each process one timeout period to finish what it is
+        doing, then kill it (SIGKILL). There's no softer SIGTERM step,
+        because that adds one join period of delay for what is almost
+        definitely either: going to behave broadly the same as to SIGKILL,
+        or won't respond to SIGTERM.
+        """
+        logger.error("BENC CLOSE1")
         for proc in self.procs:
-            proc.terminate()
+            proc.join(timeout=timeout)
+            if proc.is_alive():
+                logger.info("Usage tracking process did not end itself; sending SIGKILL")
+                proc.kill()
+        logger.error("BENC CLOSE2")
```

Comment:
> ```python
> logger.error("BENC ...")
> ```
Leftover debugging detail?

Reply:
Does the `.join(timeout=<value_greater_than_0>)` introduce a noticeable delay anywhere of consequence?  That is, it takes _some_ time for the UDP message to send and the child process to close, and unlike before, we're now waiting inline for that to happen.  (Don't get me wrong, we should absolutely join the child process!)

The first idea that comes to mind is that `.send_message()` should wrap that up automatically, since `udp_messenger` is `@async_process` and has no logic to handle more than one message at a time.  That would similarly elide the need for `.close()` at all.  On the other hand, that flirts with `threading`, which is anathema to `multiprocessing`...

Reply:
It takes 20ms on my laptop with cached DNS. When DNS is malfunctioning, it will take much longer (the classic "many second delay is always DNS") and that's what the timeout here should bound.

One argument against send message not waiting inline is to avoid that above potential timeout delay at startup, where it's much more noticeable/annoying to users: this model gives the startup message the entire length of the workflow to complete asynchronously before introducing timeout delays.

Reply:
Have talked out of band on this one -- I was suggesting another approach.  Let's not go through with it now, but it's perhaps on the mental radar of N future tasks.